### PR TITLE
[FLINK-7102] improve ClassLoaderITCase

### PR DESCRIPTION
**Pull request by NicoK**
* `ClassLoaderITCase` unnecessarily runs multiple tests in a single test case
* `ClassLoaderITCase#testDisposeSavepointWithCustomKvState()` does not cancel its
  job (thus the order of execution of test cases defines the outcome)
* `ClassLoaderITCase` uses `e.getCause().getCause()` which may cause
  `NullPointerException`s hiding the original error
* `ClassLoaderITCase` simply ignores `SuccessException` instances but does not actually verify that they are thrown/not thrown where expected

### CODE REVIEW

File: flink-tests/src/test/java/org/apache/flink/test/classloading/ClassLoaderITCase.java [hunk 1]
```diff
@@ -137,6 +137,20 @@ public void testJobsWithCustomClassLoader() throws IOException, ProgramInvocatio
 
 			inputSplitTestProg.invokeInteractiveModeForExecution();
 
+		} catch (Exception e) {
+			if (!(e.getCause() != null &&
```
Comment (pnowojski) [inline flink-tests/src/test/java/org/apache/flink/test/classloading/ClassLoaderITCase.java]: please at least wrap this `if` statement into a function `swallowScuessException(e)` or something like that.

Another idea is to maybe use this pattern:
```
	@Rule
	public ExpectedException expectedException = ExpectedException.none();

	@Test
	public void test() {
		expectedException.expectCause(isA(SuccessException.class));
		somethingThatThrows();
	}
```
( https://stackoverflow.com/a/20759785/8149051 )

**Comment by NicoK** [comment]
you're right - wasn't looking thoroughly enough (and only recently learned about `ExpectedException`).

Additionally, I found out that this ignored exception was actually required to be thrown for some of these tests while others didn't even throw it. So with the newest changes (also rebased on top of the current `master`), the tests actually better reflect the intended behaviour.

**Comment by pnowojski** [comment]
ok, I missed that you didn't fail if the expected exception didn't happen :) Now it's LGTM

**Comment by NicoK** [comment]
...that's how wrong the original implementation was - let me add this to the description of the PR

**Comment by zentol** [comment]
merging.

**Pull request closed without merging**